frame/support/src/storage/types/counted_map.rs

```diff
@@ -274,12 +275,18 @@ where
 	}
 
 	/// Remove all value of the storage.
+	#[deprecated = "Use `clear` instead"]
 	pub fn remove_all() {
+		Self::clear(None);
+	}
+
+	/// Remove all value of the storage.
+	pub fn clear(limit: Option<u32>) -> ClearPrefixResult {
 		// NOTE: it is not possible to remove up to some limit because
 		// `sp_io::storage::clear_prefix` and `StorageMap::remove_all` don't give the number of
 		// value removed from the overlay.
 		CounterFor::<Prefix>::set(0u32);
```

Comment:
this looks incorrect given that there is a limit now.

Also `remove_all` should keep this logic setting the counter to zero.

Reply:
Fixed.